controller/lib/controller.mustache

```diff
@@ -10,3 +12,17 @@ exports.normaliseParams = params => {
     return p;
   }, {});
 }
+
+// return the first query param found that is indexed (id takes precedence)
+exports.getIndex = (schema, params) => {
+  const fields = Object.keys(params);
+  if (params.id && fields['id'] && fields['id'].indexed) return 'id';
+
+  const properties = schema.properties;
+  for (var key in params) {
+    if (params.hasOwnProperty(key)) { 
+      if (properties[key] && properties[key].indexed) return key;
+    }
+  }
```

Comment:
You've added `lodash` at the top so you could take advantage of it with this loop, with something simple like:
```
_.findKey(params, (val, key) => properties[key].indexed)
```